content/docs/command-reference/exp/run.md

```diff
@@ -85,9 +85,10 @@ process finishes or gets interrupted (e.g. with Ctrl + `C`), DVC will
 <abbr>workspace</abbr> (overwriting any further changes done by the stage).
 
 `dvc exp run` again will continue from this point (useful for interrupted runs).
-Use `--reset` to roll-back the workspace to `HEAD` and restart the whole
-experiment. Alternatively, you can use `--rev` to continue from a specific
-(previous) checkpoint.
+Use `--reset` to reset (remove) any existing `checkpoint` outputs in your
+workspace before running the experiment. Note that `--reset` overrides any
```

Comment:
Sounds like "use reset to reset". We could just keep "to remove" I think. But...

> (remove) any existing `checkpoint` outputs

A little low level. What does `--reset` do from the user perspective. Does it not roll back to `HEAD` anymore?

Reply:
To be clearer, here are the steps implied in this section (checkpoints workflow with reset):

0. enable checkpoints in the stage ✅
1. update code to write signal files ✅
2. `exo run` ✅ (and possibly interrupt)
3. \> `exp run --reset` < - We are here

Why would we want to reset? Again, to restart the experiment, right? (Go back to step 2.)

Reply:
For `checkpoint` outs, `reset` removes the specified outs entirely (so that the specified intermediate model can be retrained). It used to roll back to `HEAD`, but this was changed (as of the core PR for these docs).

see https://github.com/iterative/dvc/issues/5553 for the product-side discussion on why this change was made

Reply:
I see, thanks for the context.

> For checkpoint outs

Is there another use case for `--reset`?

Reply:
How about

```md
Use `--reset` to remove `checkpoint` outputs completely to restart the
experiment from scratch regardless of any cached contents.
```

But should we move `exp run --reset` up in the steps or say it's an alternative to plain `exp run` from the get-go?

0, 1, (same as above)
2a. `exp run` - runs experiment from `HEAD` state
2b. `exp run --reset` - runs experiment from scratch

Reply:
If it's a new stage with new `checkpoint` outputs, both `2a` and `2b` are the same thing here (since there would not be any HEAD `dvc.lock` entries for the checkpoint outs).

Reply:
OK so the first `exp run` of a checkpoint pipeline is the same as `exp run --reset`, got it. So yeah this whole section needs a bit of work to include all these key points. Please address my other comments/suggestions when you can and lmk to take this over a some final reorg of ideas (I'll try it in the meantime to confirm I got it 100% now).

Reply:
One more Q @pmrowla. `exp run --reset` still deletes the checkpoints also, right? Meaning the Git commits (I'm guessing the cache is left alone).

Reply:
The experiment ref for the previously existing checkpoint run is what gets "deleted". In practice, as long as the user's stage is deterministic the old existing is just moved to point to the new run.

Git commits are left alone, git will garbage collect them on its own once they are no longer pointed to by any references.

Reply:
Noted.